universal-application-tool-0.0.1/app/assets/javascripts/azure_upload.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * This is responsible for uploading a file to Azure blob storage.
+ */
+
+class AzureUploadController {
+  static UPLOAD_CONTAINER_ID = 'azure-upload-form-component'
+
+  constructor() {
+    const uploadContainer = document.getElementById(
+      AzureUploadController.UPLOAD_CONTAINER_ID
+    )
+    uploadContainer.addEventListener('submit', (event) =>
+      this.attemptUpload(event, uploadContainer)
+    )
+  }
+
+  getValueFromInputLabel(label: string): string {
+    return (<HTMLInputElement>document.getElementsByName(label)[0]).value
+  }
+
+  attemptUpload(event: Event, uploadContainer: HTMLElement | null) {
+    event.preventDefault()
+    // This is set by an imported JavaScript Azure blob client.
+    // window["azblob"] was used because the TS compiler complained about using
+    // window.azblob.
+    const azblob = window['azblob']
+    if (uploadContainer == null) {
+      throw 'Attempted to upload to null container'
+    }
+    const azureUploadProps = this.getAzureUploadProps(uploadContainer)
+
+    const redirectUrl = new URL(azureUploadProps.successActionRedirect)
+
+    const blockBlobURL = azblob.BlockBlobURL.fromBlobURL(
+      new azblob.BlobURL(
+        `${azureUploadProps.blobUrl}?${azureUploadProps.sasToken}`,
+        azblob.StorageURL.newPipeline(new azblob.AnonymousCredential())
+      )
+    )
+
+    azblob
+      .uploadBrowserDataToBlockBlob(
+        azblob.Aborter.none,
+        azureUploadProps.file,
+        blockBlobURL
+      )
+      .then((resp, err) => {
+        if (err) {
+          throw err
+        } else {
```

Comment:
nit: no need for the `else {` here, can save a level of indentation for the rest of this function body

Reply:
Done.